### PR TITLE
PHP docs/tweaks

### PR DESCRIPTION
**Pull request by elken**
Few changes rolled into one PR, summarised below:

- Add a tutorial to help new users get started with PHP
- Make a slight correction to Intelephense to clarify it can be automatically installed
- Adjust Intelephense to use new `defcustom-lsp` macro

### CODE REVIEW

**Review by yyoncho — COMMENTED** [review]
Looks great!

Few comments:

1. There 2 files that are with size = 0  - `docs/tutorials/images/php-signature-help` and `docs/tutorials/images/php-errors-list` 
2. Can you link you as an author - like @ericdallo did here - https://github.com/yyoncho/lsp-mode/blob/master/docs/tutorials/clojure-guide.md#L8 ?
3. Can you mention somewhere that there are other language servers supporting PHP? If you have knowledge about them you may drop a line when one would prefer them.

**Comment by elken** [comment]
> Looks great!
> 
> Few comments:
> 
>     1. There 2 files that are with size = 0  - `docs/tutorials/images/php-signature-help` and `docs/tutorials/images/php-errors-list`
Completely missed those, flameshot created them by mistake and I missed them. Will remove.
>     2. Can you link you as an author - like @ericdallo did here - https://github.com/yyoncho/lsp-mode/blob/master/docs/tutorials/clojure-guide.md#L8 ?
Will do!
>     3. Can you mention somewhere that there are other language servers supporting PHP? If you have knowledge about them you may drop a line when one would prefer them.
I have at the top of the tutorial listed the 4 supported by lsp-mode

File: docs/tutorials/php-guide.md [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+- [lsp-ivy](https://github.com/emacs-lsp/lsp-ivy)
+- [consult-lsp](https://github.com/gagbo/consult-lsp)
+
+![errors list](images/php-errors-list.png)
+
+## Debugging
+
+Much like `lsp` is to code navigation/refactoring, there exists a similar spec
+called `dap` ([debug adapter
+protocol](https://microsoft.github.io/debug-adapter-protocol/)) for debugging.
```
Comment (yyoncho) [inline docs/tutorials/php-guide.md:255]: Wonder whether this will be rendered correctly with link text split into 2 lines)

Comment (elken) [reply docs/tutorials/php-guide.md:258]: It should do, but I think it's just `gq` weirdness. I'll correct it.

**Comment by ericdallo** [comment]
Looks good!

**Review by yyoncho — APPROVED** [review]

**Comment by yyoncho** [comment]
One more thing that is missing: can you link the tutorial here: https://github.com/yyoncho/lsp-mode/blob/master/mkdocs.yml#L143

**Comment by elken** [comment]
> One more thing that is missing: can you link the tutorial here: https://github.com/yyoncho/lsp-mode/blob/master/mkdocs.yml#L143

Amended

**Comment by ericdallo** [comment]
Any thoughts on why lint is failing?

**Comment by elken** [comment]
> Any thoughts on why lint is failing?

AFAICT seems to be related to the recently added perlnavigator client. I'll see if I can get it running locally in act

**Comment by yyoncho** [comment]
That lint task is failing for a long time in setup phase.

**Comment by elken** [comment]
I see, there's still a lint failure on [this](https://github.com/emacs-lsp/lsp-mode/blob/3d2b8523d57be74b0f7d4966b9ee1f9d4b118852/clients/lsp-perlnavigator.el#L54) line

**Comment by yyoncho** [comment]
@elken you are right - there are a bunch of failures in the setup phase and this line fails for the master branch.

**Comment by yyoncho** [comment]
Thank you!

**Pull request merged**